### ISSUE TITLE
list.remove(x): x not in list

### ISSUE THREAD

**Issue by Ragowit** [danielyule/hearthbreaker#108]
Deck 1: Lord of the Arena, Demolisher, Spiteful Smith, Duplicate, Darkscale Healer, Silverback Patriarch, Novice Engineer, Shattered Sun Cleric, Stonesplinter Trogg, Imp Master, Gelbin Mekkatorque, Dragonling Mechanic, Water Elemental, River Crocolisk, War Golem, Frostwolf Warlord, Azure Drake, Dancing Swords, Cult Master, Puddlestomper, Frostwolf Warlord, Stormpike Commando, Silvermoon Guardian, Frostwolf Grunt, Kel'Thuzad, Flamecannon, Wisp, Murloc Raider, Chillwind Yeti, Defender of Argus

Deck 2: Deathlord, Muster for Battle, Divine Favor, Doomsayer, Mogor the Ogre, Faceless Manipulator, Avenging Wrath, Spellbreaker, Piloted Shredder, Amani Berserker, Sneed's Old Shredder, Loot Hoarder, Mechanical Yeti, Priestess of Elune, Frost Elemental, Holy Light, Blessing of Kings, Tauren Warrior, Silver Hand Knight, Faerie Dragon, Holy Light, Ysera, Doomsayer, Stonetusk Boar, Piloted Sky Golem, Windfury Harpy, Voodoo Doctor, Humility, Leeroy Jenkins, Ancient Watcher

```
Traceback (most recent call last):
  File "C:\Users\Ragowit\Documents\GitHub\mcts\hearthbreaker\uct.py", line 162, in DoMove
    self.game._start_turn()
  File "C:\Users\Ragowit\Documents\GitHub\mcts\hearthbreaker\hearthbreaker\engine.py", line 167, in _start_turn
    self.current_player.trigger("turn_started", self.current_player)
  File "C:\Users\Ragowit\Documents\GitHub\mcts\hearthbreaker\hearthbreaker\game_objects.py", line 130, in trigger
    handler[0](*args)
  File "C:\Users\Ragowit\Documents\GitHub\mcts\hearthbreaker\hearthbreaker\tags\base.py", line 485, in _find_target
    if not tag.do(self.owner, focus, other):
  File "C:\Users\Ragowit\Documents\GitHub\mcts\hearthbreaker\hearthbreaker\tags\base.py", line 539, in do
    action.act(owner, t, other)
  File "C:\Users\Ragowit\Documents\GitHub\mcts\hearthbreaker\hearthbreaker\tags\action.py", line 201, in act
    target.die(None)
  File "C:\Users\Ragowit\Documents\GitHub\mcts\hearthbreaker\hearthbreaker\game_objects.py", line 973, in die
    self.remove_from_board()
  File "C:\Users\Ragowit\Documents\GitHub\mcts\hearthbreaker\hearthbreaker\game_objects.py", line 906, in remove_from_board
    self.player.minions.remove(self)
ValueError: list.remove(x): x not in list
```

**Comment by Ragowit** [comment]
Seems to be Poultryizer (from Gelbin Mekkatorque) that is the issue. Because it made itself into a chicken?

**Comment by Ragowit** [comment]
Nightmare (from Ysera) + Faceless Manipulator combo?

**Comment by Ragowit** [comment]
I think I know what causes the problem. It's a combo between cards like Nightmare and Poultryizer.

Nightmare will kill the minion at the start of your turn and Poultryizer will transform a minion to a chicken at the start of your turn.

If Poultryizer kicks in on the minion with Nightmare, so will Nightmare still try to kill it's minion which is no longer in play. This is because we are already in a loop (line 124 in game_objects.py) and doesn't know that Nightmare is gone.

**Comment by Ragowit** [comment]
Closed by 5d0fa537a9ec990b926e4fd8175ef2a9de06b3bf.

**Comment by Ragowit** [comment]
Had to rollback, b8c70d2e2afc81d5d31e4962548a14fe41fc7c24

**Comment by danielyule** [comment]
I managed to make a test that generated this error, and traced the source to transform() which was not properly removing the minion that was transformed.

**Comment by danielyule** [comment]
Closed in e4b88dc40a49edd5a4a918c2d74f3c887fd7c847
